src/webgpu/shader/execution/shader_io/compute_builtins.spec.ts

```diff
@@ -254,3 +255,505 @@ g.test('inputs')
       typedLength: outputBuffer.size / 4,
     });
   });
+
+/**
+ * @returns The population count of input.
+ *
+ * @param input Treated as an unsigned 32-bit integer
+ */
+function popcount(input: number): number {
+  let n = input;
+  n = n - ((n >> 1) & 0x55555555);
+  n = (n & 0x33333333) + ((n >> 2) & 0x33333333);
+  return (((n + (n >> 4)) & 0xf0f0f0f) * 0x1010101) >> 24;
+}
+
+function ErrorMsg(msg: string, got: number, expected: number): string {
+  return `${msg}:\n-      got: ${got}\n- expected: ${expected}`;
+}
+
+/**
+ * Checks that the subgroup size and ballot buffers are consistent.
+ *
+ * This function assumes subgroups will be filled as much as possible with
+ * up to 1 partial subgroup at the end of each workgroup.
+ *
+ * @param data The subgroup_size buffer
+ * @param compare The ballot buffer
+ * @param min The minimum subgroup size allowed
+ * @param max The maximum subgroup size allowed
+ * @param invocations The number of invocations in a workgroup
+ */
+function checkSubgroupSizeConsistency(
+  data: Uint32Array,
+  compare: Uint32Array,
```

Comment:
Would it be clearer to rename `data` as e.g. `subgroupSizeResults` and `compare` as `ballotSizeResults`?